core/connectors/sdk/src/decoders/proto.rs

```diff
@@ -0,0 +1,159 @@
+pub struct ProtoConfig {
+    pub schema_path: Option<String>,
+    pub message_type: Option<String>,
+    pub use_any_wrapper: bool,
+    pub field_mappings: Option<HashMap<String, String>>,
+}
+
+impl Default for ProtoConfig {
+    fn default() -> Self {
+        Self {
+            schema_path: None,
+            message_type: None,
+            use_any_wrapper: true,
+            field_mappings: None,
+        }
+    }
+}
+
+pub struct ProtoStreamDecoder {
+    config: ProtoConfig,
+}
+
+impl ProtoStreamDecoder {
+    pub fn new(config: ProtoConfig) -> Self {
+        Self { config }
+    }
+
+    pub fn new_default() -> Self {
+        Self::new(ProtoConfig::default())
+    }
+
+    fn decode_as_any(&self, payload: Vec<u8>) -> Result<Payload, Error> {
+        match Any::decode(payload.as_slice()) {
+            Ok(any) => {
+                let json_value = simd_json::json!({
+                    "type_url": any.type_url,
+                    "value": base64::engine::general_purpose::STANDARD.encode(&any.value),
+                });
+                Ok(Payload::Json(json_value))
+            }
+            Err(e) => {
+                error!("Failed to decode protobuf Any message: {e}");
+                Err(Error::CannotDecode(Schema::Proto))
+            }
+        }
+    }
+
+    fn decode_as_raw(&self, payload: Vec<u8>) -> Result<Payload, Error> {
+        Ok(Payload::Raw(payload))
+    }
+}
+
+impl StreamDecoder for ProtoStreamDecoder {
+    fn schema(&self) -> Schema {
+        Schema::Proto
+    }
+
+    fn decode(&self, payload: Vec<u8>) -> Result<Payload, Error> {
+        if payload.is_empty() {
+            return Err(Error::InvalidPayloadType);
+        }
+        println!("jarek");
+        if self.config.use_any_wrapper {
+            self.decode_as_any(payload)
+        } else {
+            self.decode_as_raw(payload)
+        }
+    }
+}
+
+impl Default for ProtoStreamDecoder {
+    fn default() -> Self {
+        Self::new_default()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use prost_types::Any;
+
+    #[test]
+    fn test_decode_protobuf_any() {
+        let decoder = ProtoStreamDecoder::new_default();
+        
+        let any = Any {
+            type_url: "type.googleapis.com/google.protobuf.StringValue".to_string(),
+            value: b"Hello, World!".to_vec(),
+        };
+        
+        let encoded = any.encode_to_vec();
+        let result = decoder.decode(encoded);
+        
+        assert!(result.is_ok());
+        if let Ok(Payload::Json(json_value)) = result {
+            if let simd_json::OwnedValue::Object(map) = &json_value {
+                assert!(map.contains_key("type_url"));
+                assert!(map.contains_key("value"));
+            } else {
+                panic!("Expected JSON object");
+            }
+        } else {
+            panic!("Expected JSON payload");
+        }
+    }
+
+    #[test]
+    fn test_decode_empty_payload() {
```

Comment:
Let's name the test methods by following the basic BDD convention e.g. `decode_should_fail_given_empty_payload` or something like this :)